src/test/java/seedu/address/storage/JsonAdaptedPoolTest.java

```diff
@@ -0,0 +1,87 @@
+
+    private static final String VALID_TRIPDAY = OFFICEPOOL.getTripDay().toString();
+    private static final String VALID_TRIPTIME = OFFICEPOOL.getTripTime().toString();
+    private static final List<JsonAdaptedPassenger> VALID_PASSENGERS = OFFICEPOOL.getPassengers().stream()
+            .map(JsonAdaptedPassenger::new)
+            .collect(Collectors.toList());
+    private static final List<JsonAdaptedTag> VALID_TAGS = OFFICEPOOL.getTags().stream()
+            .map(JsonAdaptedTag::new)
+            .collect(Collectors.toList());
+    private static final JsonAdaptedDriver VALID_DRIVER = new JsonAdaptedDriver(OFFICEPOOL.getDriver());
+
+    @Test
+    public void toModelType_validPoolDetails_returnsPool() throws Exception {
+        JsonAdaptedPool pool = new JsonAdaptedPool(HOMEPOOL);
+        assertEquals(HOMEPOOL, pool.toModelType());
+    }
+
+    @Test
+    public void toModelType_invalidTripDay_throwsIllegalValueException() {
+        JsonAdaptedPool pool =
+                new JsonAdaptedPool(INVALID_TRIPDAY, VALID_TRIPTIME, VALID_PASSENGERS, VALID_TAGS, VALID_DRIVER);
+        String expectedMessage = TripDay.MESSAGE_CONSTRAINTS;
+        assertThrows(IllegalValueException.class, expectedMessage, pool::toModelType);
+    }
+
+
+    @Test
+    public void toModelType_nullTripDay_throwsIllegalValueException() {
+        JsonAdaptedPool pool =
+                new JsonAdaptedPool(null, VALID_TRIPTIME, VALID_PASSENGERS, VALID_TAGS, VALID_DRIVER);
+        String expectedMessage = String.format(MISSING_FIELD_MESSAGE_FORMAT, TripDay.class.getSimpleName());
+        assertThrows(IllegalValueException.class, expectedMessage, pool::toModelType);
+    }
+
+    @Test
+    public void toModelType_invalidTripTime_throwsIllegalValueException() {
+        JsonAdaptedPool pool =
+                new JsonAdaptedPool(VALID_TRIPDAY, INVALID_TRIPTIME, VALID_PASSENGERS, VALID_TAGS, VALID_DRIVER);
+        String expectedMessage = TripTime.MESSAGE_CONSTRAINTS;
+        assertThrows(IllegalValueException.class, expectedMessage, pool::toModelType);
+    }
+
+    @Test
+    public void toModelType_nullTripTime_throwsIllegalValueException() {
+        JsonAdaptedPool pool =
+                new JsonAdaptedPool(VALID_TRIPDAY, null, VALID_PASSENGERS, VALID_TAGS, VALID_DRIVER);
+        String expectedMessage = String.format(MISSING_FIELD_MESSAGE_FORMAT, TripTime.class.getSimpleName());
+        assertThrows(IllegalValueException.class, expectedMessage, pool::toModelType);
+    }
+
+
+    @Test
+    public void toModelType_invalidTags_throwsIllegalValueException() {
+        List<JsonAdaptedTag> invalidTags = new ArrayList<>(VALID_TAGS);
+        invalidTags.add(new JsonAdaptedTag(INVALID_TAG));
+        JsonAdaptedPool pool =
+                new JsonAdaptedPool(VALID_TRIPDAY, VALID_TRIPTIME, VALID_PASSENGERS, invalidTags, VALID_DRIVER);
+        assertThrows(IllegalValueException.class, pool::toModelType);
+    }
```

Comment:
i guesss this is okay, but some of these tests could be refactored out and put under one umbrella test method that tests invalid tags. Within the test they are seperated by comments // that tells the reader what the tests are

Reply:
The point of the tests is to test the whole chain when reading a JsonAdaptedPool, I think there is any issue here.